### PR TITLE
Feature/ad user interaction

### PR DESCRIPTION
**Pull request by DaniilAndCo**
adding ability to users to create and maintain ads, using migration for db version control and project structure refactoring

### CODE REVIEW

**Comment by DaniilAndCo** [comment]
waiting for review on done things

**Comment by DaniilAndCo** [comment]
the next things I am gonna implement:
-ability to assign make of a vihicle in the right way
-storing profile and ads photos and manage this

File: src/main/java/com/github/daniilandco/vehicle_sales_project/config/SecurityConfig.java [hunk 1]
```diff
@@ -16,8 +16,7 @@
 @EnableWebSecurity
 @EnableGlobalMethodSecurity(prePostEnabled = true)
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
-
-    private static int encryptionStrength = 12;
+    private static final int encryptionStrength = 12;
```
Comment (asenchuk) [inline src/main/java/com/github/daniilandco/vehicle_sales_project/config/SecurityConfig.java:19]: if it's a constant it should be named like this `ENCRYPTION_STRENGTH`

File: src/main/java/com/github/daniilandco/vehicle_sales_project/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    @GetMapping("/ads/{id}")
+    public ResponseEntity<?> getAdById(@PathVariable Long id) {
+        return adServiceImplementation.getUserAdById(id);
+    }
+
+    @PostMapping("/add")
+    public ResponseEntity<?> addAd(@RequestBody NewAdRequest request) {
+        return adServiceImplementation.addAd(request);
+    }
```
Comment (asenchuk) [inline src/main/java/com/github/daniilandco/vehicle_sales_project/controller/UserController.java:38]: why not moving ad-related endpoints to `AdController` with a base context `/ads`?

**Review by asenchuk — CHANGES_REQUESTED** [review]

**Comment by asenchuk** [comment]
CategoryRequest seems to be not added to the repo

**Pull request merged**